core/Macros.carp

```diff
@@ -724,6 +724,9 @@ The expression must be evaluable at compile time.")
 (doc *gensym-counter* "is a helper counter for `gensym`.")
 (defdynamic *gensym-counter* 1000)
 
+(defndynamic gensym-local [x]
```

Comment:
This does not prevent accidental shadowing, correct?

Reply:
That's correct, no shadowing prevention here. If you wind up using this in a context in which some `gensym-generatedX` already exists (where `x` is whatever you pass as an argument) the resulting generated symbol will shadow the extant one--I guess we could get around this by adding another symbol to the end--something like `gensym-generated-localX`?